### PR TITLE
Add minimum distance for pilot breaking

### PR DESCRIPTION
**Pull request by metalgearsloth**
:cl:
- tweak: Piloting now breaks on minimum distance (0.25 units) and not any distance

### CODE REVIEW

**Review by PaulRitter — COMMENTED** [review]
why not just take the distance from the console to the pilot?

**Comment by 20kdc** [comment]
Presumably that'd allow being dragged all the way behind the console and leave generally a lot of room for movement. 0.25 units wouldn't be enough then.

**Comment by metalgearsloth** [comment]
Could do 0.5 from centre and then that leaves 0.2 of movement? (when considering 0.25 extent for console and 0.175 for player)

File: Content.Server/Shuttles/ShuttleConsoleSystem.cs [hunk 1]
```diff
@@ -72,7 +73,10 @@ private void HandlePowerChange(EntityUid uid, ShuttleConsoleComponent component,
         /// </summary>
         private void HandlePilotMove(EntityUid uid, PilotComponent component, ref MoveEvent args)
         {
-            if (component.Console == null) return;
+            if (component.Console == null ||
+                args.NewPosition.TryDistance(EntityManager, component.Position!.Value, out var distance) &&
```
Comment (PaulRitter) [inline Content.Server/Shuttles/ShuttleConsoleSystem.cs:77]: just handle the `component.Position` sanely tbh (as in removing the pilot if its null)

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by PaulRitter** [comment]
> 
> 
> Could do 0.5 from centre and then that leaves 0.2 of movement? (when considering 0.25 extent for console and 0.175 for player)

eh, i suppose its fine the way it is.
image

File: Content.Server/Shuttles/ShuttleConsoleSystem.cs [hunk 2]
```diff
@@ -72,7 +73,16 @@ private void HandlePowerChange(EntityUid uid, ShuttleConsoleComponent component,
...
         {
-            if (component.Console == null) return;
+            if (component.Console == null)
+            {
+                DebugTools.Assert(component.Position == null);
+                EntityManager.RemoveComponent<PilotComponent>(uid);
+                return;
+            }
+
+            if (args.NewPosition.TryDistance(EntityManager, component.Position!.Value, out var distance) &&
```
Comment (PaulRitter) [inline Content.Server/Shuttles/ShuttleConsoleSystem.cs:83]: mfw the nullable is still not handled sanely
image

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: Content.Server/Shuttles/ShuttleConsoleSystem.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Server/Shuttles/ShuttleConsoleSystem.cs:83]: Listen alright, if it's ever null at that stage then the server deserved what it got

**Comment by metalgearsloth** [comment]
I also realised if remote piloting makes it in doing it this way is potentially better than checking to the console.

**Pull request merged**